src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compFunc =
+      compareFunction ||
+      function (a, b) {
+        const aStr = String(a);
+        const bStr = String(b);
+
+        if (aStr < bStr) {
+          return -1;
+        }
+
+        if (aStr > bStr) {
+          return 1;
+        }
+
+        return 0;
+      };
```

Comment:
The default compare function is correctly implemented for handling the case where no compareFunction is provided. However, you might want to consider extracting this default function outside of the sort2 method for better readability and potential reuse.